docs: add spa mode section in rendering (#27426)
nuxt/nuxt: docs/2.guide/1.concepts/3.rendering.md

:read-more{title="SPA Loading Template" to="/docs/api/configuration/nuxt-config#spaloadingtemplate"}
::

> ### SPA Mode

Reviewer: ```suggestion
### Deploying a Static Client-Rendered App
```

> Nuxt provides the ability to deploy your application into a [static hosting](getting-started/deployment#static-hosting). But when you use `ssr: false` and run `nuxi generate`, it will produce all the pages as static HTML files. In a common SPA we only need one single `index.html` file. In order to achieve this we can change how the routes are prerendered. Just add this to your hooks in your `nuxt.config.ts`:

Reviewer: ```suggestion
If you deploy your app to [static hosting](getting-started/deployment#static-hosting) with the `nuxi generate` or `nuxi build --prerender` commands, then by default, Nuxt will render every page as a separate static HTML file.

If you are using purely client-side rendering, then this might be unnecessary. You might only need a single `index.html` file, plus `200.html` and `404.html` fallbacks, which you can tell your static web host to serve up for all requests.

In order to achieve this we can change how the routes are prerendered. Just add this to your hooks in your `nuxt.config.ts`:
```

```ts twoslash [nuxt.config.ts]
export default defineNuxtConfig({
  hooks: {

>     "prerender:routes"({ routes }) {

Reviewer: ```suggestion
    'prerender:routes' ({ routes }) {
```

:read-more{title="SPA Loading Template" to="/docs/api/configuration/nuxt-config#spaloadingtemplate"}
::

> 

Reviewer: ```suggestion

::tip{to="https://www.youtube.com/watch?v=7Lr0QTP1Ro8" icon="i-logos-youtube-icon" target="_blank"}
Watch a video from Alexander Lichter about **Building a plain SPA with Nuxt!?**.
::

```

### Deploying a Static Client-Rendered App

> If you deploy your app to [static hosting](getting-started/deployment#static-hosting) with the `nuxi generate` or `nuxi build --prerender` commands, then by default, Nuxt will render every page as a separate static HTML file.

Reviewer 2: Could possible also add a video alert on top to https://www.youtube.com/watch?v=7Lr0QTP1Ro8 🤔

Reviewer: Good idea - happy to make the code suggestion?

> If you deploy your app to [static hosting](getting-started/deployment#static-hosting) with the `nuxi generate` or `nuxi build --prerender` commands, then by default, Nuxt will render every page as a separate static HTML file.

Reviewer: ```suggestion
If you deploy your app to [static hosting](/docs/getting-started/deployment#static-hosting) with the `nuxi generate` or `nuxi build --prerender` commands, then by default, Nuxt will render every page as a separate static HTML file.
```

If you are using purely client-side rendering, then this might be unnecessary. You might only need a single `index.html` file, plus `200.html` and `404.html` fallbacks, which you can tell your static web host to serve up for all requests.

> In order to achieve this we can change how the routes are prerendered. Just add this to your hooks in your `nuxt.config.ts`:

Reviewer: ```suggestion
In order to achieve this we can change how the routes are prerendered. Just add this to [your hooks](/docs/api/advanced/hooks#nuxt-hooks-build-time) in your `nuxt.config.ts`:
```

```ts twoslash [nuxt.config.ts]
export default defineNuxtConfig({
  hooks: {
    'prerender:routes' ({ routes }) {

>       routes.clear() // Do not generate any routes (except the defaults)

Reviewer 2: Would be nice to link to the Nuxt hooks or mention them somewhere close by.
